src/shared/ui/inputs/Input/Input.tsx

```diff
@@ -1,46 +1,22 @@
-import { twJoin } from "tailwind-merge";
+import * as React from "react";
 
-interface InputProps {
-  icon?: React.ReactNode;
-  id: string;
-  label: string;
-  placeholder: string;
-  className?: string;
-  onChange?: React.ChangeEventHandler<HTMLInputElement>;
-}
+interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {}
 
-export function Input({
-  icon,
-  id,
-  label,
-  className,
-  placeholder,
-  onChange,
-}: InputProps) {
-  return (
-    <div className={twJoin("text-color-base", className)}>
-      <label htmlFor={id} className="block text-sm font-bold">
-        {label}
-      </label>
+/**
+ * Renders a basic html input component
+ */
 
-      <div className="relative mt-spacing-tighter-2 rounded shadow-sm">
-        <input
-          type="text"
-          name={id}
-          id={id}
-          className={twJoin(
-            "block min-h-control-size-base w-full rounded-border-radius-base border border-essential-subdued bg-background-base py-spacing-tighter-2 font-semibold shadow-essential-base placeholder:text-color-subdued hover:border-essential-base focus:border-essential-base focus:border-white focus:ring-white",
-            icon && "pr-[calc(24px+var(--spacing-tighter)*2)]",
-          )}
-          placeholder={placeholder}
-          onChange={onChange}
-        />
-        {icon && (
-          <div className="pointer-events-none absolute inset-y-0 right-0 flex items-center pr-spacing-tighter">
-            {icon}
-          </div>
-        )}
-      </div>
-    </div>
-  );
-}
+export const Input = React.forwardRef<HTMLInputElement, InputProps>(
+  function Input({ className, onChange, placeholder, type, ...props }, ref) {
+    return (
+      <input
+        {...props}
+        ref={ref}
+        className={className}
+        onChange={onChange}
+        placeholder={placeholder}
+        type={type}
```

Comment:
I think we can remove them. They are included in the `{...props}`. Is is useful to declare them when you want to add default behaviour for a props. For example for className:

```
twJoin(`bg-red-500`, className)
```

> Ok, by default the element has a red background, but I allow consumers to override it or add more classes if needed

Reply:
Good point, I deleted them except classname, as you can see below.